test/ICSharpCode.SharpZipLib.Tests/Zip/ZipEncryptionHandling.cs

```diff
@@ -434,40 +541,77 @@ public void ZipFileAESReadWithEmptyPassword()
+            GnLy8zwsuV0Jh/RQSwMEMwABAGMAnWNsUgAAAAAhAAAABQAAAAkACwBGaWxlMi50eHQBmQcAAgBBRQMAANoCDbQUG7iCJgGC2/5OrmUQUk/+fACL804W0bboF8YMM1BLAwQzAA
+            EAYwCjY2xSAAAAACEAAAAFAAAACQALAEZpbGUzLnR4dAGZBwACAEFFAwAAqmBqgBkl3tP6ND0uCD50mhwfOtbmwV1IKyUVK5wGVQUiUEsBAj8AMwABAGMAj2NsUgAAAAAhAAAA
+            BQAAAAkALwAAAAAAAAAgAAAAAAAAAEZpbGUxLnR4dAoAIAAAAAAAAQAYAApFb9MyF9cB7fEh1jIX1wHNrjnNMhfXAQGZBwACAEFFAwAAUEsBAj8AMwABAGMAnWNsUgAAAAAhAA
+            AABQAAAAkALwAAAAAAAAAgAAAAUwAAAEZpbGUyLnR4dAoAIAAAAAAAAQAYAK5pWOQyF9cBdTCL5TIX1wGab3HVMhfXAQGZBwACAEFFAwAAUEsBAj8AMwABAGMAo2NsUgAAAAAh
+            AAAABQAAAAkALwAAAAAAAAAgAAAApgAAAEZpbGUzLnR4dAoAIAAAAAAAAQAYANB1M+kyF9cB0gxl6jIX1wGqVSHWMhfXAQGZBwACAEFFAwAAUEsFBgAAAAADAAMAMgEAAPkAAAAAAA==";
 
 		/// <summary>
-		/// ZipInputStream can't decrypt AES encrypted entries, but it should report that to the caller
-		/// rather than just failing.
+		/// Test reading an AES encrypted file and skipping some entries by not reading form the stream
 		/// </summary>
 		[Test]
 		[Category("Zip")]
-		public void ZipinputStreamShouldGracefullyFailWithAESStreams()
+		public void ZipFileAESReadSkippingEntriesIsPossible()
 		{
-			string password = "password";
+			var fileBytes = Convert.FromBase64String(TestFileWithThreeEntries);
 
-			using (var memoryStream = new MemoryStream())
+			using (var ms = new MemoryStream(fileBytes))
+			using (var zis = new ZipInputStream(ms) { IsStreamOwner = false})
 			{
-				// Try to create a zip stream
-				WriteEncryptedZipToStream(memoryStream, password, 256);
+				zis.Password = "password";
 
-				// reset
-				memoryStream.Seek(0, SeekOrigin.Begin);
-
-				// Try to read
-				using (var inputStream = new ZipInputStream(memoryStream))
+				for (int i = 0; i < 3; i++)
 				{
-					inputStream.Password = password;
-					var entry = inputStream.GetNextEntry();
-					Assert.That(entry.AESKeySize, Is.EqualTo(256), "Test entry should be AES256 encrypted.");
-
-					// CanDecompressEntry should be false.
-					Assert.That(inputStream.CanDecompressEntry, Is.False, "CanDecompressEntry should be false for AES encrypted entries");
+					var entry = zis.GetNextEntry();
+					if (i == 1)
+					{
+						continue;
+					}
 
-					// Should throw on read.
-					Assert.Throws<ZipException>(() => inputStream.ReadByte());
+					using (var sr = new StreamReader(zis, Encoding.UTF8, leaveOpen: true, detectEncodingFromByteOrderMarks: true, bufferSize: 1024))
+					{
+						var content = sr.ReadToEnd();
+						Assert.AreEqual(Path.GetFileNameWithoutExtension(entry.Name), content, "Decompressed content does not match input data");
+					}
 				}
 			}
 		}
+		
+		/// <summary>
+		/// Test reading an AES encrypted file and reading some entries only partially be not reading to the end of stream.
+		/// </summary>
+		[Test]
+		[Category("Zip")]
+		public void ZipFileAESReadEntriesCanBeReadPartially()
```

Comment:
```suggestion
		public void ZipInputStreamAESReadEntriesCanBeReadPartially()
```
Same as above, `ZipFile` is another class in the library.
I guess the naming is because it's testing on a static precomputed file?